[MIG] hr_employee_firstname: Migration to 17.0 t#76430

andreagidaltig opened the pull request:
[Task #76430](https://www.vauxoo.com/web#id=76430&cids=1&menu_id=1530&model=project.task&view_type=form)
- Update definition of required modifier as part of [1].
- Update incoming parameters of post_init_hook method since `cr, registry`
  were replaced by `env` in [2].
- Deprecate the use of _onchange_spec method and use the method new method
  _get_fields_spec to return the fields specification from a view
  description since it was introduced in [3].
  
[1] https://github.com/odoo/odoo/pull/104741
[2] https://github.com/odoo/odoo/commit/b4a7996
[3] https://github.com/odoo/odoo/commit/f5e6494

feketemihai committed:
[ADD] hr_employee_firstname

Commit:
Remove field name related, change init script, add unittests

Commit:
[FIX+IMP] hr_employee_firstname: Several things:

* remove ensure one
* Add decorator
* Update name on write and update related partner firstname lastname
* Modify README file
* Change version format
* Consistency with partner_firstname module
* Check if the partner isn't already in the list
* Remove check on partner_firstname installation

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[IMP][hr_employee_firstname] Avoid to update partners without reason

Commit:
[FIX] Update tests in hr_employee_firstname - failing travis

feketemihai committed:
[REF] hr_employee_firstname: Split names at module install to get the value not the key of dict (#215)

* Update split names at module install to get the value not the key of the dict.
* Add test for checking right values in firstname, lastname after install

espo-tony committed:
[MIG] hr_employee_firstname: Migrated to 10.0 (#276)

Commit:
[MIG] hr_employee_firstname: Migration to version 11.0

feketemihai committed:
[FIX] employee_firstname: test (#427)

ddufresne committed:
[11.0][FIX] Fix dangerous use of 'replace' from hr.employee view.

This issue was flagged by travis in the following build https://travis-ci.org/OCA/hr/jobs/364219109

oca-transbot committed:
OCA Transbot updated translations from Transifex

fshahy committed:
Translated using Weblate (Persian)

Currently translated at 100.0% (3 of 3 strings)

Translation: hr-11.0/hr-11.0-hr_employee_firstname
Translate-URL: https://translation.odoo-community.org/projects/hr-11-0/hr-11-0-hr_employee_firstname/fa/

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (3 of 3 strings)

Translation: hr-11.0/hr-11.0-hr_employee_firstname
Translate-URL: https://translation.odoo-community.org/projects/hr-11-0/hr-11-0-hr_employee_firstname/de/

astirpe committed:
[12.0][MIG] hr_employee_firstname

astirpe committed:
[FIX/IMP] Make required attributes dynamic

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (4 of 4 strings)

Translation: hr-12.0/hr-12.0-hr_employee_firstname
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_employee_firstname/de/

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (4 of 4 strings)

Translation: hr-12.0/hr-12.0-hr_employee_firstname
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_employee_firstname/es/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (4 of 4 strings)

Translation: hr-12.0/hr-12.0-hr_employee_firstname
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_employee_firstname/pt_BR/

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (4 of 4 strings)

Translation: hr-12.0/hr-12.0-hr_employee_firstname
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_employee_firstname/de/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (4 of 4 strings)

Translation: hr-12.0/hr-12.0-hr_employee_firstname
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_employee_firstname/hr/

astirpe committed:
[13.0][MIG] hr_employee_firstname

astirpe committed:
pre-commit

OCA-git-bot committed:
pre-commit update

Saran440 committed:
[13.0][FIX] hr_employee_firstname : testscript

astirpe committed:
[IMP] hr_employee_firstname: ease extension

OCA-git-bot committed:
hr_employee_firstname 13.0.1.0.1

luistorresm committed:
[MIG] hr_employee_firstname: Migration to 14.0

Commit:
[REF] hr_employee_firstname: Removed partner_firstname dependency

OCA-git-bot committed:
[IMP] update dotfiles [ci skip]

OCA-git-bot committed:
hr_employee_firstname 14.0.2.0.0

luistorresm committed:
[FIX] hr_employee_firstname: Avoid access error on employee update

``sudo`` is used to avoid access error when verify that module is installed.

Avoid error in the next case:
1. Create a new user without settings access (HR User)
2. Login with the user in 1, and try to edit an employee

OCA-git-bot committed:
hr_employee_firstname 14.0.2.0.1

britoederr committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (4 of 4 strings)

Translation: hr-14.0/hr-14.0-hr_employee_firstname
Translate-URL: https://translation.odoo-community.org/projects/hr-14-0/hr-14-0-hr_employee_firstname/pt_BR/

Commit:
Added translation using Weblate (Catalan)

hugho-ad committed:
[FIX] hr_employee_firstname: Move fields to common employee

Starting from 14.0, employee model was split into three models:
- `hr.employee.public`
- `hr.employee.private`
- `hr.employee` (a common model)

When this module was migrated, such  split was not taken into account.

This commit moves fields to the common model, so they don't fail when
read from the public employee model.

Closes #990

Co-Authored-By: Luis González <lgonzalez@vauxoo.com>

OCA-git-bot committed:
hr_employee_firstname 14.0.3.0.0

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (11 of 11 strings)

Translation: hr-14.0/hr-14.0-hr_employee_firstname
Translate-URL: https://translation.odoo-community.org/projects/hr-14-0/hr-14-0-hr_employee_firstname/de/

luistorresm committed:
[MIG] hr_employee_firstname: Migration to 15.0

douglascstd committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 54.5% (6 of 11 strings)

Translation: hr-15.0/hr-15.0-hr_employee_firstname
Translate-URL: https://translation.odoo-community.org/projects/hr-15-0/hr-15-0-hr_employee_firstname/pt_BR/

luistorresm committed:
[MIG] hr_employee_firstname: Migration to 16.0

Commit:
[UPD] Update hr_employee_firstname.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: hr-16.0/hr-16.0-hr_employee_firstname
Translate-URL: https://translation.odoo-community.org/projects/hr-16-0/hr-16-0-hr_employee_firstname/

luistorresm committed:
[FIX] hr_employee_firstname: Improved create on employee

Avoid WARNING on inherit to create method:
```
The model odoo.addons.hr_employee_firstname.models.hr_employee is not
overriding the create method in batch
```

Commit:
[UPD] Update hr_employee_firstname.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: hr-16.0/hr-16.0-hr_employee_firstname
Translate-URL: https://translation.odoo-community.org/projects/hr-16-0/hr-16-0-hr_employee_firstname/

Commit:
[UPD] Update hr_employee_firstname.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: hr-16.0/hr-16.0-hr_employee_firstname
Translate-URL: https://translation.odoo-community.org/projects/hr-16-0/hr-16-0-hr_employee_firstname/

FernandoRomera committed:
[16.0][FIX] test refresh->invalidate_recordset

OCA-git-bot committed:
hr_employee_firstname 16.0.1.0.1

mymage committed:
Translated using Weblate (Italian)

Currently translated at 87.5% (7 of 8 strings)

Translation: hr-16.0/hr-16.0-hr_employee_firstname
Translate-URL: https://translation.odoo-community.org/projects/hr-16-0/hr-16-0-hr_employee_firstname/it/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (8 of 8 strings)

Translation: hr-16.0/hr-16.0-hr_employee_firstname
Translate-URL: https://translation.odoo-community.org/projects/hr-16-0/hr-16-0-hr_employee_firstname/es/

OCA-git-bot committed:
[UPD] README.rst

andreagidaltig committed:
[IMP] hr_employee_firstname: rename hooks file

Apply guidelines on name of hooks file.

OCA-git-bot committed:
[BOT] post-merge updates

andreagidaltig commented:
Hi @luistorresm could you review this please?

luistorresm commented:
Hello @andreagidaltig 

Could you join the commits related, please?

More info [here](https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate)

cc @luisg123v

andreagidaltig commented:
Hi @luistorresm 

I verified this with @luisg123v  and I just merged some translation commits from same author and same language as is indicated [here](https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate).

luistorresm commented:
👍 

@luisg123v 

Could you merge, please?

Regards

luisg123v commented on hr_employee_firstname/tests/test_hr_employee_firstname.py:
```diff
@@ -0,0 +1,180 @@
+        """
+        Validate the get_name method is not failing
+        """
+        field_onchange = self.env["hr.employee"].new({})._get_fields_spec()
```
This is not an onchange anymore AFAIK, so I think you shouldn't neither name it as so nor using `new()`.

luisg123v commented on hr_employee_firstname/tests/test_hr_employee_firstname.py:
```diff
@@ -0,0 +1,180 @@
+        field_onchange = self.env["hr.employee"].new({})._get_fields_spec()
+
+        self.assertEqual(
+            submap(field_onchange, ('firstname', 'lastname')),
```
Why this?

luisg123v commented:
Missing task ID on PR's title.

andreagidaltig committed:
[MIG] hr_employee_firstname: Migration to 17.0

- Update definition of required modifier as part of [1].
- Update incoming parameters of post_init_hook method since `cr, registry`
  were replaced by `env` in [2].
- Deprecate the use of _onchange_spec method and use the method new method
  _get_fields_spec to return the fields specification from a view
  description since it was introduced in [3].

[1] https://github.com/odoo/odoo/pull/104741
[2] https://github.com/odoo/odoo/commit/b4a7996
[3] https://github.com/odoo/odoo/commit/f5e6494

andreagidaltig commented:
Hi @luisg123v I just modified this, could you review it again please?

cc. @luistorresm

luisg123v reviewed (approved):
LGTM :+1:

luisg123v commented:
Be aware once the OCA releases their branches, this will get overwritten and we will need to make a new PR targeting the OCA.
